### PR TITLE
backend: dwarfdbginf: add UTF encoding to dchar types

### PR DESCRIPTION
**Pull request by ljmf00**
Currently `dchar` types are exported as signed encoding which is wrong, because
`dchar` is supposed to represent UTF-32 encoded characters, so we should export
the right encoding to the debug info.

Signed-off-by: Luís Ferreira <contact@lsferreira.net>

### CODE REVIEW

**Comment by thewilsonator** [comment]
tests?

**Comment by ljmf00** [comment]
> tests?

I'm going to add some

**Comment by Luhrel** [comment]
Same problem for `TYwchar_t`, but `DW_AT_encoding` is not emitted at all.

**Comment by ljmf00** [comment]
> Same problem for `TYwchar_t`, but `DW_AT_encoding` is not emitted at all.

Yes, although this needs to be addressed on another PR because the current `wchar_t` type mimics the C behaviour and since this backend is shared with DMC, some extra care is needed. See https://github.com/dlang/dmd/pull/13247 for more context.

File: test/dshell/extra-files/dwarf/fix22468.d [hunk 1]
```diff
@@ -0,0 +1,5 @@
+/*
+EXTRA_ARGS: -g -defaultlib= -betterC -c
```
Comment (ljmf00) [inline test/dshell/extra-files/dwarf/fix22468.d:2]: I'm compiling with those flags to make the compiler not generate other types other than `dchar`.

**Comment by ljmf00** [comment]
semaphoreci seems to be failing for unrelated reasons but it is GDC related so I'm not sure what is going on there.

**Comment by RazvanN7** [comment]
I've restarted semaphore, let's see what happens.

**Review by PetarKirov — APPROVED** [review]
LGTM. For reference:

image

(https://www.dwarfstd.org/doc/DWARF5.pdf, page 340)

**Comment by PetarKirov** [comment]
The Semaphore failure should be a fluke, as all 3 builds (based on the same DMD code) have different results:

1. All green
2. GDC failing, everything else is green
3. LDC failing, everything else is green

I'm going to rebase on master both to restart the build and to test with the latest state of things.

**Pull request merged**